lib/browser-specific.js

```diff
@@ -0,0 +1,256 @@
+'use strict';
+
+/**
+ * Implements functionality to report on how many WPT tests fail only on one
+ * browser (aka browser-specific failures).
+ */
+
+// A helper class providing an iterator-like interface to either an Object with
+// enumerable properties, or an Array. Iterates in a sorted order determined by
+// |comparatorFunc|.
+class IteratorHelper {
+  constructor(arrOrObject, comparatorFunc) {
+    this.currentIndex = 0;
+    this.values = arrOrObject;
+
+    if (Array.isArray(this.values)) {
+      this.keys = null;
+      this.maxIndex = this.values.length;
+      this.values.sort(comparatorFunc);
```

Comment:
This mutates the array passed in, which fine, but not what you'd expect from an iterator.